kratos/processes/from_json_check_result_process.h

```diff
@@ -0,0 +1,925 @@
+
+    /**
+     * @brief This method returns the nodes of the model part
+     * @return The nodes of the model part
+     */
+    NodesArrayType& GetNodes(const Flags* pFlag = nullptr);
+
+    /**
+     * @brief This method returns the elements of the model part
+     * @return The elements of the model part
+     */
+    ElementsArrayType& GetElements(const Flags* pFlag = nullptr);
+
+    /**
+     * @brief This method computes the relevant digits to take into account
+     */
+    std::size_t ComputeRelevantDigits(const double Value);
+
+    /**
+     * @brief This method provides the defaults parameters to avoid conflicts between the different constructors
+     */
+    const Parameters GetDefaultParameters() const override;
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    /**
+     * @brief This method returns the model part
+     * @return The model part of the problem
+     */
+    ModelPart& GetModelPart() const;
+
+    /**
+     * @brief This method returns the settings
+     * @return The settings of the problem
+     */
+    Parameters GetSettings() const;
+
+    /**
+     * @brief This method returns the Nodes database. If not initialized it will try initialize again
+     * @return The nodes database
+     */
+    const ResultDatabase& GetNodeDatabase();
+
+    /**
+     * @brief This method returns the GP database. If not initialized it will try initialize again
+     * @return The GP database
+     */
+    const ResultDatabase& GetGPDatabase();
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    /// Protected constructor with modified default settings to be defined by derived class.
+    FromJSONCheckResultProcess(ModelPart& rModelPart, Parameters Settings, Parameters DefaultSettings);
+
+    ///@}
+
+private:
+    ///@name Member Variables
+    ///@{
+
+    /* Model part and different settings */
+    ModelPart& mrModelPart;         /// The main model part
+    Parameters mThisParameters;     /// The parameters (can be used for general pourposes)
+
+    /* Additional values */
+    double mFrequency;              /// The check frequency
+    double mRelativeTolerance;      /// The relative tolerance
+    double mAbsoluteTolerance;      /// The absolute tolerance
+    std::size_t mRelevantDigits;    /// This is the number of relevant digits
+
+    /* Counters */
+    double mTimeCounter = 0.0;      /// A time counter
+
+    /* The entities of the containers */
+    NodesArrayType mNodesArray;       /// The nodes of study
+    ElementsArrayType mElementsArray; /// The elements of study
+
+    /* The vectors storing the variables of study */
+    std::vector<const Variable<double>*> mpNodalVariableDoubleList;                     /// The scalar variable list to compute
+    std::vector<const VariableComponent<ComponentType>*> mpNodalVariableComponentsList; /// The scalar variable list to compute (components) // TODO: To be replaced once components are unified
```

Comment:
I would wait with this until #6686 is merged

Reply:
Indeed (but if I want a clean implementation i need to add this)